test/manipulation.test.js

```diff
@@ -756,54 +756,65 @@ describe('manipulation', function () {
     it('should destroy filtered set of records');
   });
 
-  describe('deleteAll/destroyAll', function () {
-    beforeEach(function clearOldData(done) {
-      Person.deleteAll(done);
+  describe('when deleteAll/destroyAll executes', function () {
+    it('should be defined as function', function() {
+      Person.deleteAll.should.be.a.Function;
+      Person.destroyAll.should.be.a.Function;
     });
 
-    beforeEach(function createTestData(done) {
-      Person.create([
-        { name: 'John' },
-        { name: 'Jane' }
-      ], done);
-    });
+    describe('with multiple instances in the database', function() {
+      beforeEach(function deleteFixtures(done) {
+        Person.deleteAll(done);
+      });
 
-    it('should only delete instances that satisfy the where condition', function (done) {
-      Person.deleteAll({name: 'John'}, function (err, result) {
-        if (err) return done(err);
-        result.should.have.property('count', 1);
-        Person.find({where: {name: 'John'}}, function (err, data) {
+      beforeEach(function createFixtures(done) {
+        Person.create([{
+          name: 'John'
+        }, {
+          name: 'Jane'
+        }], done);
+      });
+
+      it('should only delete instances that satisfy the where condition',
+          function(done) {
+        Person.deleteAll({name: 'John'}, function(err, info) {
           if (err) return done(err);
-          data.should.have.length(0);
-          Person.find({where: {name: 'Jane'}}, function (err, data) {
+          info.count.should.equal(1);
```

Comment:
Actually when I look at this again, we are modifying the second arg in this case to return `{count: 0}` instead, meaning `data.should.have.length(0)` won't work since its not an array being returned anymore. Any suggestions for this?